File: codeup/100000576/C.py
#!/usr/bin/python3
import sys


class student_info:
    def __init__(self, id, name, sex, age):
        self.id = id
        self.name = name
        self.sex = sex
        self.age = age


while True:
    line = sys.stdin.readline()
    if not line:
        break
    si_dist = {}
    N = int(line)
    for i in range(N):
        id, name, sex, age = input().split()
        si_dist[id] = student_info(id, name, sex, age)

    M = int(input())
    for i in range(M):
        id = input()
        if id in si_dist:
            si_rst = si_dist[id]
            print(si_rst.id, si_rst.name, si_rst.sex, si_rst.age)
        else:
            print("No Answer!")

File: codeup/100000576/C.cpp
#include <cstring>
#include <iostream>
using namespace std;

struct student {
  char id[10];
  char name[100];
  char sex[5];
  int age;
} student_list[1020];

int main() {
  int N, M, i, j;

  while (scanf("%d", &N) != EOF) {
    for (i = 0; i < N; i++) {
      scanf("%s %s %s %d", student_list[i].id, student_list[i].name,
            student_list[i].sex, &student_list[i].age);
    }
    scanf("%d", &M);
    char id[10];
    for (i = 0; i < M; i++) {
      scanf("%s", id);
      for (j = 0; j < N; j++) {
        if (strcmp(id, student_list[j].id) == 0) break;
      }
      if (j < N)
        printf("%s %s %s %d\n", student_list[j].id, student_list[j].name,
               student_list[j].sex, student_list[j].age);
      else
        printf("No Answer!\n");
    }
  }

  return 0;
}
